### PR TITLE
Prevent null deref crash in WebrtcVideoConduit DTOR

### LINKED ISSUE #1558: Null deref crash in webrtc ~WebrtcVideoConduit
You can check it on the Basilisk only because webrtc disabled in Palemoon.
file - media\webrtc\signaling\src\media-conduit\VideoConduit.cpp

~WebrtcVideoConduit() has SyncTo(nullptr); which is coded as
```

   mPtrViEBase->DisconnectAudioChannel(mChannel);
   mPtrViEBase->SetVoiceEngine(nullptr);
```

meanwhile DeleteStreams() has

`mPtrViEBase = nullptr;`

and executed earlier ~WebrtcVideoConduit(), so you get excpetion 0xc0000005 in result
I moved SyncTo(nullptr) to DeleteStreams() before mPtrViEBase = nullptr and bug seems gone

### PR DESCRIPTION
**Pull request by Feodor2**
mPtrViEBase was used after being set to nullptr

This resolves #1558

### CODE REVIEW

File: media/webrtc/signaling/src/media-conduit/VideoConduit.cpp [hunk 1]
```diff
@@ -551,8 +552,10 @@ WebrtcVideoConduit::SyncTo(WebrtcAudioConduit *aConduit)
     mPtrViEBase->ConnectAudioChannel(mChannel, aConduit->GetChannel());
     // NOTE: this means the VideoConduit will keep the AudioConduit alive!
   } else {
+    if (mPtrViEBase) {
```
Comment (wolfbeast) [inline media/webrtc/signaling/src/media-conduit/VideoConduit.cpp:555]: No.
If you're going to have a guard here, it should be an early return, not dependent on `aConduit`

File: media/webrtc/signaling/src/media-conduit/VideoConduit.cpp [hunk 2]
```diff
@@ -523,6 +523,7 @@ WebrtcVideoConduit::DeleteStreams()
   mVideoCodecStat = nullptr;
   // We can't delete the VideoEngine until all these are released!
   // And we can't use a Scoped ptr, since the order is arbitrary
+  SyncTo(nullptr);
```
Comment (wolfbeast) [inline media/webrtc/signaling/src/media-conduit/VideoConduit.cpp:526]: Please move this up above the comment that relates to nulling the pointers, and give it a comment of its own

Comment (Feodor2) [reply media/webrtc/signaling/src/media-conduit/VideoConduit.cpp]: Done as you wanted

**Comment by mattatobin** [comment]
Please follow the contribution guidelines regarding proper commit titles and redo your patch with the changes included.

Then force push to your pr branch.

**Comment by mattatobin** [comment]
Now can you please clean it up again?

**Comment by Feodor2** [comment]
Seems done right

**Comment by mattatobin** [comment]
Is now ye. Hopefully there aren't any other issues and we can get this merged.

**Pull request merged**